invenio_search/__init__.py

```diff
@@ -22,7 +22,257 @@
+Now we can finally create the index.
+
+.. code-block:: console
+
+   $ export FLASK_APP=app.py
+   $ flask index init
+
+If you are running Elasticsearch on your local machine, you can see that the
+indices were created correctly by going to this URL:
+http://localhost:9200/demo
+
+Let's put some data inside
+
+.. code-block:: console
+
+   $ echo '{"title": "Hello invenio-search", "body": "test 1"}' | \
+         flask index put demo-default-v1.0.0 example
+
+No error message? Good! You can see that your new document was indexed by
+going to this URL: http://localhost:9200/demo/_search
+
+Searching for data
+~~~~~~~~~~~~~~~~~~
+
+Let's try to retrieve data in a programmatic way. Start a Python REPL and run
+the following commands.
+
+First, let's initialize app from the ``app.py`` file that we created in the previous step.
+Python REPL.
+
+.. code-block:: python
+
+   from app import app
+
+Create a custom search class that will search for ``example`` type of
+documents inside the ``demo`` index (or you could use the default
+``RecordsSearch`` class to search for all document types in all indexes):
+
+.. code-block:: python
+
+   from invenio_search import RecordsSearch
+   class ExampleSearch(RecordsSearch):
+       class Meta:
+           index = 'demo'
+           doc_types = ['example']
+           fields = ('*', )
+           facets = {}
+   search = ExampleSearch()
+
+We will need a Flask application context, so let's push one:
+
+.. code-block:: python
+
+   app.app_context().push()
+
+Let's find all documents:
+
+.. code-block:: python
+
+   response = search.execute()
+   response.to_dict()
+
+If everything went well, you should now see that we have 1 hit - a document
+with ``Hello invenio-search`` title. If you get the ``TransportError(404, 'index_not_found_exception', 'no such index')`` error - it means that you
+forgot to create the index (follow the steps from :ref:`creating_index` to
+see how to setup an index and add example data).
+
+
+Creating a search page
+~~~~~~~~~~~~~~~~~~~~~~
+
+Let's create a simple web page where you can send queries to the Elasticsearch
+and see the results.
+Create a new app.py file with a route.
+
+.. code-block:: python
+
+    # app.py
+    from elasticsearch_dsl.query import QueryString
+    from flask import Flask, jsonify, request
+    from invenio_search import InvenioSearch, RecordsSearch
+
+    app = Flask('myapp')
+
+    search = InvenioSearch(app)
+
+    # This line is needed to be able to call `flask index init`
+    search.register_mappings('demo', 'examples.data.elastic_v2')
+
+
+    @app.route('/', methods=['GET', 'POST'])
+    def index():
+        search = RecordsSearch()
+        if 'q' in request.values:
+            search = search.query(QueryString(query=request.values.get('q')))
+
+        return jsonify(search.execute().to_dict())
+
+Run example development server:
+
+.. code-block:: console
+
+   $ FLASK_APP=app.py flask run --debugger -p 5000
```

Comment:
``FLASK_DEBUG=1 FLASK_APP=app.py flask run`` is the "new" way of doing it. Not sure ``--debugger`` works anymore.